cmd/clusterctl/client/tree/discovery.go

```diff
@@ -112,8 +112,10 @@ func Discovery(ctx context.Context, c client.Client, namespace, name string, opt
 				tree.Add(m, machineInfra, ObjectMetaName("MachineInfrastructure"), NoEcho(true))
 			}
 
-			if machineBootstrap, err := external.Get(ctx, c, m.Spec.Bootstrap.ConfigRef, cluster.Namespace); err == nil {
-				tree.Add(m, machineBootstrap, ObjectMetaName("BootstrapConfig"), NoEcho(true))
+			if m.Spec.Bootstrap.ConfigRef != nil {
```

Comment:
The infraRef isn't a pointer so I don't think it'd be strictly necessary as it would just fail silently. But we could add a check for `infraRef == corev1.ObjectReference{}` for good measure.